### PR TITLE
REFACTOR: TMA-1198 Set global logger in LoggerMiddleware instead of GoodDataMiddleware

### PR DESCRIPTION
**Pull request by sadam21**

### CODE REVIEW

File: spec/unit/bricks/middleware/logger_middleware_spec.rb [hunk 1]
```diff
@@ -49,7 +50,9 @@
       it 'sets info log level' do
         expect(logger).to receive(:level=).with('info')
         subject.call(params)
+        expect(logger).to eq(GoodData.logger)
       end
     end
   end
+
```
Comment (yenkins) [inline spec/unit/bricks/middleware/logger_middleware_spec.rb]: warning: Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body end. (rubocop)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/898/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1711/ : SUCCESS in 1m 12s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1972/ : FAILURE in 1m 12s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/149/ : SUCCESS in 7m 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/899/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1712/ : SUCCESS in 1m 11s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1973/ : SUCCESS in 1m 15s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/150/ : SUCCESS in 7m 27s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/900/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1713/ : SUCCESS in 1m 23s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1974/ : SUCCESS in 1m 20s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/151/ : SUCCESS in 7m 32s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/152/ : SUCCESS in 14s

**Pull request merged**